pkg/testworkflows/testworkflowexecutor/executor.go

```diff
@@ -542,11 +532,55 @@ func (e *executor) Execute(ctx context.Context, workflow testworkflowsv1.TestWor
 		log.DefaultLogger.Errorw("failed to encode tags", "id", id, "error", err)
 	}
 
+	// Get (for centralized mode) TW execution or create it
+	if request.Id != "" {
```

Comment:
We need to de-concrete executions logic from here to split fully executor engine from data implementation